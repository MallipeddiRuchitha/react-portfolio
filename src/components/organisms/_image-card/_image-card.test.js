import React from 'react';
import { shallow ,mount} from 'enzyme';
import ImageCard from './_image-card';
import EditIcon from "@material-ui/icons/Edit";
import DeleteIcon from "@material-ui/icons/Delete";
const image={id:1,name:"forest1",url:"https://www.birmingham.ac.uk/Images/News/Forest-900.jpg"}

describe("ImageCard component",() =>{
    test("Matches the snapshot",() => {
        const imageCard = shallow(<ImageCard image1={image} />);
        expect(imageCard).toMatchSnapshot();

    })
    test("Image Card  element is rendered",() => {
        const imageCard = mount(<ImageCard image1={image} />);
        expect(imageCard.find('#imagecard').exists()).toEqual(true);
        //console.log(imageCard.debug());
        
    });
    test("edit icon element is rendered",() => {
        const imageCard = shallow(<ImageCard image1={image} />);
        expect(imageCard.find(EditIcon).exists()).toEqual(true);        
    });
    
    test("delete icon element is rendered",() => {
        const imageCard = shallow(<ImageCard image1={image} />);
        expect(imageCard.find(DeleteIcon).exists()).toEqual(true);
    });
    
    test("on click of DeleteIcon",() => {
        const mockOnClick = jest.fn();
        const imageCard = shallow(<ImageCard image1={image} handleDelete={mockOnClick}/>);
        const deleteIcon=imageCard.find(DeleteIcon);
        deleteIcon.simulate('click');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
    test("Editing the ImageCard",() => {
        //const galleryAdmin= shallow(<GalleryAdmin/>);        
       //galleryAdmin.find(EditIcon).first().simulate('click');
       //expect(galleryAdmin.find(EditImageForm).exists()).toEqual(true);
         
    });
    



});