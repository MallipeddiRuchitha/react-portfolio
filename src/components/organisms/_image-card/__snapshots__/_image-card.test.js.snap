// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleImageCard-1"
  id="imagecard"
>
  <div
    className="makeStyles-picture-2"
  >
    <Image
      image="https://www.birmingham.ac.uk/Images/News/Forest-900.jpg"
    />
  </div>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-name-3"
  >
    image
  </WithStyles(ForwardRef(Typography))>
  <div
    className="makeStyles-icons-5"
  >
    <EditIcon
      className="makeStyles-edit-4"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
    <DeleteIcon
      id="deleteIcon"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
  </div>
</div>
`;
